tests/drivers/flash/boards/disco_l475_iot1.overlay

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2022 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&flash0 {
```

Comment:
To run test on flash using that board, you also need to update chosen `chosen {zephyr,flash-controller = &mx25r6435f;};`
But then, we need to be sure that quadspi flash controller is tested with other tests.